IanGRap/Squish
# Lobby Color Change Rework

Contributors: Alfredo, Austin

Goals:
- Get rid of paint blocks (and code)
- Create new system for changing color on Dab
- Reorganize lobby hierarchy
- Fix the font bug in Begin Rave screen.
- Changed "Begin Rave" to "Ready to Rave?" in the UI

IMPORTANT NOTE:
- Right now, we don't have a finished Y button icon. We will make a tiny PR to put it in once the asset is finished.

What Changed:
- Got rid of PaintBlockCollision, PlayerSelectBlockSpawner, PaintBlockMovement
- Added a new script called PlayerColorChanger, which changes player colors and spawns the visual effects
- Changed PlayerSelectColorController so that it ignores players' preferred materials on the color change dab
- Added a property called CanTaunt to PlayerTaunt.cs so that the color changer does not initiate the effect if the player is already dabbing
- Reorganized scene hierarchy
- Changed UI text and font in the Begin Match scene
- New prefab called Color Change Beam which is the spotlight visual effect
- New particle effect called Color Change Particle Effect

How to Test:
- Load up the Lobby scene
- Make sure the hierarchy is clean (before all of the title screen objects were spread all over the hierarchy rather than being consolidated to a parent object)
- Make sure that when your player is in the lobby, you can change colors with Y. 
    - You cannot change colors if you are already in the animation
    - You cannot change colors if you are on the ready stage
    - You cannot change colors when you are in the air (the conditions are the same as the dab conditions)
- Make sure that the icon colors are right in the Ready to Rave scene
- Make sure everything still works in a build

- Do you like the VFX?
